src/benchmarklib/tpch/tpch_query_generator.cpp

```diff
@@ -52,6 +71,330 @@ std::string TPCHQueryGenerator::get_preparation_queries() const {
 }
 
 std::string TPCHQueryGenerator::build_query(const QueryID query_id) {
+  using namespace std::string_literals;
+
+  static std::random_device random_device;
+  // Preferring a fast random engine over one with high-quality randomness
+  static std::minstd_rand random_engine(random_device());
+
+  static std::vector materials{"TIN", "NICKEL", "BRASS", "STEEL", "COPPER"};
+
+  // Random distributions for all strings defined by the TPC-H benchmark
+  static std::uniform_int_distribution<> material_dist{0, static_cast<int>(materials.size() - 1)};
+  static std::uniform_int_distribution<> region_dist{0, regions.count - 1};
+  static std::uniform_int_distribution<> segment_dist{0, c_mseg_set.count - 1};
+  static std::uniform_int_distribution<> nation_dist{0, nations.count - 1};
+  static std::uniform_int_distribution<> type_dist{0, p_types_set.count - 1};
+  static std::uniform_int_distribution<> color_dist{0, colors.count - 1};
+  static std::uniform_int_distribution<> shipmode_dist{0, l_smode_set.count - 1};
+  static std::uniform_int_distribution<> brand_char_dist{1, 5};
+  static std::uniform_int_distribution<> container_dist{0, p_cntr_set.count - 1};
+
+  // This is not nice, but initializing this statically would require external methods and make it harder to
```

Comment:
anonymously-namespaced or function-static C-arrays are totally fine for me, in case you'd like that better.

Reply:
Would you prefer an array over the vector and if yes, why?

Reply:
No particular preference on my side. I read your comment as implying dissatisfaction with having to use static vectors here and wanted to point out the alternative. But maybe I am misreading the comment?

Reply:
Ah, now I get it, I think! This comment mostly refers to `sizes` and not to `materials`, right?